test/cqlpy/test_map_key_selection.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2020-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+
+#############################################################################
+# Tests for ...
+#############################################################################
+
+import pytest
+import re
+import time
+from cassandra.protocol import InvalidRequest
+from .util import unique_name, unique_key_string
+
+
+@pytest.fixture(scope="module")
+def table1(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int PRIMARY KEY, m map<int, int>)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+@pytest.fixture(scope="module")
+def table2(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int PRIMARY KEY, m map<text, int>)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+@pytest.fixture(scope="module")
+def table3(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int PRIMARY KEY, m map<text, frozen<map<text, set<int>>>>)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+def test_basic_int_key_selection(cql, table1):
+    cql.execute("INSERT INTO " + table1 + "(p,m) VALUES (0, {1:10,2:20})")
+    cql.execute("INSERT INTO " + table1 + "(p,m) VALUES (1, {1:11,2:21,3:31})")
+    cql.execute("INSERT INTO " + table1 + "(p,m) VALUES (2, {3:32})")
+    assert set(cql.execute(f"SELECT p,m[1] FROM {table1}")) == {(0,10),(1,11),(2,None)}
+    assert set(cql.execute(f"SELECT p,m[2] FROM {table1}")) == {(0,20),(1,21),(2,None)}
+    assert set(cql.execute(f"SELECT p,m[3] FROM {table1}")) == {(0,None),(1,31),(2,32)}
+
+def test_basic_string_key_selection(cql, table2):
```

Comment:
Interesting. This test checks `frozen<map<text, set<int>>>`, so it also checks two more things you didn't check above - that the "frozen" case works too, and that the set<> case works too. Maybe you should add simpler tests just for this before you get into this "recursive" (actually, "nested" is the better word) case.

A small note about the fixtures and the schemas: If you have a different schema for each test, you don't need to put it into a fixture (like table7) - you can use new_test_table() directly in the tests - we have many tests doing that. A fixture is good if you can use the same schema for multiple tests. That's actually possible - you can create a single table with several different columns, and use a different column in every test. This will help shave a few milliseconds from the test run, because fewer tables need to be created and destroyed :-)

Reply:
I added a case with frozen map, and did some cleanup of the test fixtures